packages/front-end/components/Settings/UpgradeModal/SelfHostedUpgradeForm.tsx

```diff
@@ -0,0 +1,271 @@
+import { useEffect, useState } from "react";
+import { useUser } from "../../../services/UserContext";
+import { useForm } from "react-hook-form";
+import fetch from "node-fetch";
+import track from "../../../services/track";
+import Field from "../../Forms/Field";
+import LoadingSpinner from "../../LoadingSpinner";
+import { FaExclamationTriangle, FaRegCheckCircle } from "react-icons/fa";
+
+const LICENSE_KEY_API_URL = "https://license.growthbook.io/api";
+
+export default function SelfHostedUpgradeForm({
+  source,
+  setCloseCta,
+  close,
+}: {
+  source: string;
+  setCloseCta: (string) => void;
+  close: () => void;
+}) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [submitState, setSubmitState] = useState<
+    false | "send key success" | "resend success" | "resend failed"
+  >(false);
+  const [useResendForm, setUseResendForm] = useState(false);
+
+  const { accountPlan, name, email, organization } = useUser();
+
+  useEffect(() => {
+    track("View Upgrade Modal", {
+      accountPlan,
+      source,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const form = useForm({
+    defaultValues: {
+      name,
+      email,
+      seats: Math.max(organization?.members?.length || 10, 10),
```

Comment:
Use same algorithm as back-end (`getNumberOfUniqueMembersAndInvites`)

Reply:
The API should store this number in Mongo, but max out the seats in the generated license key

Reply:
Also send `source` and `reason` and `dateCreated` and current `build` to the API